From 3ec514cf260b82f4701b6fed521ce470d9faf8c9 Mon Sep 17 00:00:00 2001
From: Gary Thomas <gary@mlbassoc.com>
Date: Tue, 6 Oct 2015 09:32:22 -0600
Subject: [PATCH] kernel-module-imx-gpu-viv: Better work-around for change in name of busfreq-imx.h

The renaming of include/linux/busfreq-imx6.h to include/linux/busfreq-imx.h
is not consistent over the many kernel versions currently being built.  This
changeset works around this inconsistency by creating a local symbolic link
to whatever file is actually present in the kernel sources.

Signed-off-by: Gary Thomas <gary@mlbassoc.com>
Upstream-status: Innapropriate [requires OE recipe support]
----
Index: kernel-module-imx-gpu-viv-5.0.11.p7.1/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c
===================================================================
--- kernel-module-imx-gpu-viv-5.0.11.p7.1.orig/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c
+++ kernel-module-imx-gpu-viv-5.0.11.p7.1/kernel-module-imx-gpu-viv-src/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_imx6q14.c
@@ -74,11 +74,8 @@
 #include <linux/pm_runtime.h>
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 10, 0)
 #include <mach/busfreq.h>
-#elif LINUX_VERSION_CODE < KERNEL_VERSION(3, 14, 0)
-#include <linux/busfreq-imx6.h>
-#include <linux/reset.h>
 #else
-#include <linux/busfreq-imx.h>
+#include "busfreq-imx.h"
 #include <linux/reset.h>
 #endif
 #endif
